x/ccv/consumer/keeper/validators.go

```diff
@@ -100,13 +102,74 @@ func (k Keeper) ValidatorByConsAddr(sdk.Context, sdk.ConsAddress) stakingtypes.V
 	return stakingtypes.Validator{}
 }
 
+func (k Keeper) SetLargestSoftOptOutValidatorPower(ctx sdk.Context, power uint64) {
+	store := ctx.KVStore(k.storeKey)
+	store.Set(types.SoftOptOutThresholdPowerKey(), sdk.Uint64ToBigEndian(power))
+}
+
+// UpdateLargestSoftOptOutValidatorPower sets the largest validator power that is allowed to soft opt out
+// This is the largest validator power such that the sum of the power of all validators with a lower or equal power
+// is less than 5% of the total power of all validators
+func (k Keeper) UpdateLargestSoftOptOutValidatorPower(ctx sdk.Context) {
+	// get all validators
+	valset := k.GetAllCCValidator(ctx)
+
+	// sort validators by power ascending
+	sort.Slice(valset, func(i, j int) bool {
+		return valset[i].Power < valset[j].Power
+	})
+
+	// get total power in set
+	totalPower := sdk.ZeroDec()
+	for _, val := range valset {
+		totalPower = totalPower.Add(sdk.NewDecFromInt(sdk.NewInt(val.Power)))
+	}
+
+	// get power of the biggest validator who is allowed to soft opt out
+	powerSum := sdk.ZeroDec()
+	for _, val := range valset {
+		powerSum = powerSum.Add(sdk.NewDecFromInt(sdk.NewInt(val.Power)))
+		// if powerSum / totalPower > SoftOptOutThreshold
+		if powerSum.Quo(totalPower).GT(sdk.MustNewDecFromStr(k.GetSoftOptOutThreshold(ctx))) {
+			// set largestSoftOptOutValidatorPower to the power of this validator
+			k.SetLargestSoftOptOutValidatorPower(ctx, uint64(val.Power))
+			k.Logger(ctx).Info("largest soft opt out validator power updated", "power", val.Power)
+			return
+		}
+	}
+	// This will be hit if the SoftOptOutThreshold param is equal to 1
```

Comment:
Why is this even allowed? The `SoftOptOutThreshold` param must be validated.

Reply:
It's a panic so not allowed, I've updated the comment https://github.com/cosmos/interchain-security/commit/df7ddaf2335ffe1e18c5d2018c8d2f13d8815212